src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -603,7 +603,7 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
             if (isPublicRoomResult(result)) {
                 const clientRoom = cli.getRoom(result.publicRoom.room_id);
                 const listener = (ev) => {
-                    viewRoom(result.publicRoom.room_id, true, ev.type !== "click");
+                    viewRoom(result.publicRoom.canonical_alias ?? result.publicRoom.room_id, true, ev.type !== "click");
```

Comment:
The first arg of `viewRoom` is called `roomId` yet you're giving it an alias? It then fires it using Action.ViewRoom in the `room_id` field, that must be in the `room_alias` field or the right things won't happen.

Reply:
You should use `getDisplayAliasForAliasSet(room.canonical_alias, room.aliases)` to obtain the alias for a room.

Reply:
That's a customization for rendering aliases only, hence the *Display* part. Some instances use this to remove the server part of the alias, using this for API requests would then break room joining entirely.